hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/supervisor/workload/WorkTracker.java

```diff
@@ -0,0 +1,128 @@
+            // ignore
+        } catch (Exception ex) {
+            throw new InternalProcessingException(ex);
+        }
+    }
+
+    public WorkDistributionChanges apply(SubscriptionAssignmentView targetView) {
+        SubscriptionAssignmentView currentView = getAssignments();
+
+        List<SubscriptionAssignment> deletions = currentView.deletions(targetView).getAllAssignments();
+        List<SubscriptionAssignment> additions = currentView.additions(targetView).getAllAssignments();
+
+        deletions.forEach(this::dropAssignment);
+        additions.forEach(this::addAssignment);
+
+        return new WorkDistributionChanges(deletions.size(), additions.size());
+     }
+
+    private void dropAssignment(SubscriptionAssignment assignment) {
+        askCuratorPolitely(() -> curatorClient.delete().guaranteed().forPath(pathSerializer.serialize(assignment.getSubscriptionName(), assignment.getConsumerNodeId())));
+    }
+
+    private void addAssignment(SubscriptionAssignment assignment) {
+        askCuratorPolitely(() -> curatorClient.create().creatingParentsIfNeeded().withMode(PERSISTENT).forPath(pathSerializer.serialize(assignment.getSubscriptionName(), assignment.getConsumerNodeId())));
+    }
+
+    private Set<SubscriptionAssignment> getAssignments(String subscriptionName) {
+        return getEntry(subscriptionName).getCurrentData().stream()
+                .map(child -> pathSerializer.deserialize(child.getPath())).collect(Collectors.toSet());
+    }
+
+    public SubscriptionAssignmentView getAssignments() {
+        return new SubscriptionAssignmentView(getSubcacheKeySet().stream().collect(toMap(SubscriptionName::fromString, this::getAssignments)));
+    }
+
+    public boolean isAssignedTo(SubscriptionName subscription, String consumerNodeId) {
+        return getAssignments(subscription.toString()).stream().filter(assignment ->
+                Objects.equals(assignment.getConsumerNodeId(), consumerNodeId)).findAny().isPresent();
+    }
+
+    interface CuratorTask {
+        void run() throws Exception;
+    }
+
+    @Override
+    protected SubscriptionAssignmentRegistry createSubcache(String path) {
+        return new SubscriptionAssignmentRegistry(
+                curatorClient,
+                path,
+                executorService,
+                subscriptionRepository,
+                consumerNodeId,
+                pathSerializer);
+    }
+
+    public static class WorkDistributionChanges {
+        private final int deleted;
+        private final int created;
+
+        public WorkDistributionChanges(int deleted, int created) {
+            this.deleted = deleted;
+            this.created = created;
+        }
+
+        public int getDeletedAssignmentsCount() {
+            return deleted;
+        }
+
+        public int getCreatedAssignmentsCount() {
+            return created;
+        }
+
+        public String printStats() {
```

Comment:
A bit of nitpicking: how about overloading `toString()` for that? Or at least let's not call it printing, because what we actually do is getting `formattedStats` or getting `statsAsString`:)

Same goes for `WorkBalancingResult`.